src/main/java/io/netty/buffer/api/unsafe/UnsafeMemoryManagers.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api.unsafe;
+
+import io.netty.buffer.api.MemoryManager;
+import io.netty.buffer.api.MemoryManagers;
+import io.netty.util.internal.PlatformDependent;
+
+public class UnsafeMemoryManagers implements MemoryManagers {
+    public UnsafeMemoryManagers() {
+        if (!PlatformDependent.hasUnsafe()) {
```

Comment:
I think this should go in the `static` block. WDYT?

Reply:
These instances are service loaded, so I'll have to check how that interacts… currently the provider will throw, so it can be handled when they are iterated.

Reply:
The only difference this does is that the exception thrown from the provider will have an additional ExceptionInInitializerError cause level. So I don't think it makes sense to make that change.